extends Control

@onready var continue_button: TextureButton = $VBoxContainer/ContinueButton
@onready var new_game_button: TextureButton = $VBoxContainer/NewGameButton
@onready var load_game_button: TextureButton = $VBoxContainer/LoadGameButton
@onready var settings_button: TextureButton = $VBoxContainer/SettingsButton
@onready var quit_button: TextureButton = $VBoxContainer/QuitButton

func _ready():
	if LoadingScreen.is_loading == true:
		LoadingScreen.hide_loading()
	continue_button.pressed.connect(on_continue_pressed)
	new_game_button.pressed.connect(on_new_game_pressed)
	load_game_button.pressed.connect(on_load_game_pressed)
	settings_button.pressed.connect(on_settings_pressed)
	quit_button.pressed.connect(on_quit_pressed)

# üü¢ –ü–†–û–î–û–í–ñ–ò–¢–ò –ì–†–£ (–æ—Å—Ç–∞–Ω–Ω—è)
func on_continue_pressed():
	LoadingScreen.show_loading()
	await get_tree().create_timer(2).timeout
	SaveManager.continue_game()

# üü° –ù–û–í–ê –ì–†–ê
func on_new_game_pressed():
	# –≤—ñ–¥–∫—Ä–∏–≤–∞—î–º–æ —Å—Ü–µ–Ω—É –≤–∏–±–æ—Ä—É —Å–≤—ñ—Ç—É –¥–ª—è –Ω–æ–≤–æ—ó –≥—Ä–∏
	SaveManager.set_game_mode(true)
	get_tree().change_scene_to_file("res://scenes/world/world_select_ui.tscn")

# üîµ –ó–ê–í–ê–ù–¢–ê–ñ–ò–¢–ò –ì–†–£
func on_load_game_pressed():
	# –≤—ñ–¥–∫—Ä–∏–≤–∞—î–º–æ —Å—Ü–µ–Ω—É –≤–∏–±–æ—Ä—É —Å–≤—ñ—Ç—É/–∑–±–µ—Ä–µ–∂–µ–Ω–Ω—è
	SaveManager.set_game_mode(false)
	get_tree().change_scene_to_file("res://scenes/world/world_select_ui.tscn")

# ‚öôÔ∏è –ù–ê–õ–ê–®–¢–£–í–ê–ù–ù–Ø
func on_settings_pressed():
	# –ó–∞–≤–∞–Ω—Ç–∞–∂—É—î–º–æ —Å—Ü–µ–Ω—É –Ω–∞–ª–∞—à—Ç—É–≤–∞–Ω—å
	var settings_scene = load("res://scenes/world/settings_ui.tscn")
	var settings_instance = settings_scene.instantiate()

	# –í–∏–∫–ª–∏–∫–∞—î–º–æ —Ñ—É–Ω–∫—Ü—ñ—é –≤ —Å—Ü–µ–Ω—ñ –Ω–∞–ª–∞—à—Ç—É–≤–∞–Ω—å (—è–∫—â–æ —î —Ç–∞–∫–∞ —Ñ—É–Ω–∫—Ü—ñ—è)
	settings_instance.is_main_menu(true)

	# –ó–∞–º—ñ–Ω—é—î–º–æ –ø–æ—Ç–æ—á–Ω—É —Å—Ü–µ–Ω—É –Ω–∞ –Ω–∞–ª–∞—à—Ç—É–≤–∞–Ω–Ω—è
	get_tree().root.add_child(settings_instance)
	get_tree().current_scene = settings_instance


# üî¥ –í–ò–•–Ü–î
func on_quit_pressed():
	get_tree().quit()
